context/src/main/java/io/opentelemetry/context/LazyStorage.java

```diff
@@ -79,11 +79,18 @@ static ContextStorage createStorage(AtomicReference<Throwable> deferredStorageFa
 
     List<ContextStorageProvider> providers = new ArrayList<>();
     for (ContextStorageProvider provider : ServiceLoader.load(ContextStorageProvider.class)) {
+      if (provider
+          .getClass()
+          .getName()
+          .equals("io.opentelemetry.sdk.testing.context.SettableContextStorageProvider")) {
```

Comment:
Could we create an interface for this, rather than relying on the classname explicitly?

Reply:
That would add a public API for this testing use case. I'd rather keep this implementation detail for now and if another use case comes up can add an interface for a "forced context storage provider" - wdyt?

Reply:
Agreed for now. I think it could be a confusing addition to the public API. I'm fine with this for now...especially as we can start trying it out and seeing how well it works.